dolphin/alert_manager/trap_receiver.py

```diff
@@ -95,6 +97,25 @@ def _extract_oid_value(var_bind):
 
         return oid, val
 
+    @staticmethod
+    def _get_alert_source_by_host(source_ip):
+        """Gets alert source for given source ip address."""
+        filters = {'host': source_ip}
+        ctxt = context.get_admin_context()
+
+        alert_source = db.alert_source_get_all(ctxt, filters=filters)
```

Comment:
Is there possibility of db throwing exception? Could not get session or query error..SQLAlchemyError is caught?

Reply:
All our filter apis do not through exception, returns empty list incase of failures

Reply:
Actually the DB will throw exception and that DB should handle..Please open Issue to track

Reply:
you need to cath, becasue there are filters params. There could be invalid params.
But if there is no filters, then it's ok to not catch exception

Reply:
Done